### PR TITLE
DOC: use latest numpydoc (#919)

### LINKED ISSUE #919: remove custom numpydoc distro and `.. shownumpydoc` hack
`.. shownumpydoc` is currently used to show the numpy doc for the dunder (i.e., ``__<whatever>__``) methods, but these tend to clutter the API web docs (see *Methods* [here](http://scikit-bio.org/docs/0.2.3/generated/skbio.sequence.BiologicalSequence.html#skbio.sequence.BiologicalSequence), and prevent us from using the latest and greatest ``numpydoc`` (e.g., the new ``Yields`` section). We should drop the `.. shownumpydoc` and the customized numpydoc distro. 

Once this change goes in, we should update to the latest ``numpydoc``.

### PR DESCRIPTION
**Pull request by jairideout**
Updated to use the latest numpydoc package (v0.6), removing our custom numpydoc distro and the `.. shownumpydoc` hack. Docs build much faster now: on my system, it takes 1 minute instead of 3 minutes.

Additional changes:

- updated most of `skbio.sequence` and `skbio.tree` to use numpydoc's new `Yields` section (this is for documenting generators). If devs think it's important I can update the entire codebase, but since scikit-bio is undergoing rapid changes right now, it probably makes more sense to update the rest of the docstrings as we tackle specific subpackages.

- updated our documentation system README with latest instructions

- removed some unnecessary `Attributes` sections from classes that didn't need them

Fixes #919.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ install:
   - conda create --yes -n env_name python=$PYTHON_VERSION pip numpy$NUMPY_VERSION scipy matplotlib$MATPLOTLIB_VERSION pandas nose pep8 Sphinx=1.2.2 IPython
   - if [ ${USE_CYTHON} ]; then conda install --yes -n env_name cython; fi
   - source activate env_name
-  - pip install sphinx-bootstrap-theme future six coveralls natsort pyflakes flake8 python-dateutil 'CacheControl[FileCache]'
+  - pip install sphinx-bootstrap-theme future six coveralls natsort pyflakes flake8 python-dateutil 'CacheControl[FileCache] git+git://github.com/numpy/numpydoc.git'
```
Comment (ebolyen) [inline .travis.yml]: I think you missed some single quotes here.

Comment (jairideout) [inline .travis.yml]: fixed, thanks!

**Comment by jairideout** [comment]
Tests are passing, this is ready for review @ebolyen or @anderspitman

File: doc/source/conf.py [hunk 2]
```diff
@@ -1,24 +1,45 @@
...
 from skbio.util import classproperty
 
-# NOTE: parts of this file were taken from scipy's doc/source/conf.py. See
-# scikit-bio/licenses/scipy.txt for scipy's license.
-
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-sys.path.insert(0, os.path.abspath('../sphinxext/numpydoc'))
+#sys.path.insert(0, os.path.abspath('../sphinxext/numpydoc'))
```
Comment (anderspitman) [inline doc/source/conf.py]: Can this be removed?

Comment (jairideout) [inline doc/source/conf.py]: fixed

**Comment by anderspitman** [comment]
Reviewed. Everything looks good to me.

**Pull request merged**